Week3/Homework/Project/Credit-Card-Validator.js

```diff
@@ -0,0 +1,90 @@
+function cardChecker(cardNumber) {
+
+    if (cardNumber * 10 % 2 !== 0) {        // Check if cardNumber is not a number (if it has letters or symbols)
+
+        const newCardNumber = prompt("Invalid Characters, Please Enter Just Numbers :");
+        return cardChecker(newCardNumber);
+
+    }
+
+    else if (cardNumber * 10 % 2 === 0) {       // Check if cardnumber is a number
+
+        const copyOfCardNumber = cardNumber;
+
+        if (copyOfCardNumber.length !== 16) {
+
+            const newCardNumber = prompt("The card number that you entered has not 16 digits, Please Enter a Valid Card Number :");
+            return cardChecker(newCardNumber);
+
+        }
+
+        else if (copyOfCardNumber.length === 16) {      // Check if copyOfCardNumber has 16 digits
+
+            for (let i = 1; i < copyOfCardNumber.length; i++) {     // This loop compare the first digit of copyOfCardNumber with the all other 15 digits 
+
+                if (copyOfCardNumber[0] === copyOfCardNumber[i]) {
+
+                    if (i === copyOfCardNumber.length - 1) {    // Check if it reached to the last digit of copyOfCardNumber
+
+                        const newCardNumber = prompt("The card number that you entered has the same 16 digits, Please Enter a Valid Card Number :");
+                        return cardChecker(newCardNumber);
+
+                    }
+
+                }
+
+                else {
+
+                    break;
+
+                }
+
+            }
+
+            if (copyOfCardNumber[copyOfCardNumber.length - 1] % 2 !== 0) {      // Check if the last digit of copyOfCardNumber is ood
```

Comment:
👍 but you can also divide the whole number by 2 to check if the last digit is even or odd. 😉